src/protocol/mod.rs

```diff
@@ -199,11 +200,7 @@ impl Protocol {
             .expect_field("arguments")
             .and_then(|arguments| arguments.expect_str())
         {
-            self.arguments = parser::Parser {
-                original: arguments.to_string(),
-                input: arguments.chars().collect()
-            }
-            .parse_script_arguments()?
+            self.arguments = Parser::parse_script_arguments(arguments)?
```

Comment:
here's an example of how the changes to the `parse_*` methods affect the call sites. I think this takes some of the complexity out of these calls, since we no longer have to worry about the interior structure of the parsing struct, and can simply pass a `&str` to receive our result.